File: typescript/base-hmi-types/src/screens/controls/HmiAlarmFilterExpression.ts
import { HmiAlarmFilterExpressionType } from "./HmiAlarmFilterExpressionType.js";
import { HmiAlarmFilterFieldType } from "./HmiAlarmFilterFieldType.js";
import { HmiAlarmFilterOperator } from "./HmiAlarmFilterOperator.js";
import { HmiAlarmFilterValueType } from "./HmiAlarmFilterValueType.js";

export class HmiAlarmFilterExpression {
  type = HmiAlarmFilterExpressionType.Unknown;
  field = HmiAlarmFilterFieldType.Unknown;
  sourceField?: string;
  operator = HmiAlarmFilterOperator.Unknown;
  sourceOperator?: string;
  valueType = HmiAlarmFilterValueType.Unknown;
  textValue?: string;
  numericValue?: number;
  booleanValue?: boolean;
  sourceValue?: string;
  readonly children: HmiAlarmFilterExpression[] = [];
}

File: src/BaseHmiTypes/Screens/Controls/HmiAlarmFilterExpression.cs
namespace BaseHmiTypes.Screens.Controls;

public sealed class HmiAlarmFilterExpression
{
    public HmiAlarmFilterExpressionType Type { get; set; }

    public HmiAlarmFilterFieldType Field { get; set; }

    public string? SourceField { get; set; }

    public HmiAlarmFilterOperator Operator { get; set; }

    public string? SourceOperator { get; set; }

    public HmiAlarmFilterValueType ValueType { get; set; }

    public string? TextValue { get; set; }

    public double? NumericValue { get; set; }

    public bool? BooleanValue { get; set; }

    public string? SourceValue { get; set; }

    public IList<HmiAlarmFilterExpression> Children { get; } = new List<HmiAlarmFilterExpression>();
}
